parsers/bibtex.c

```diff
@@ -0,0 +1,459 @@
+/*
+ *	 Copyright (c) 2008, David Fishburn
+ *	 Copyright (c) 2012, Jan Larres
+ *	 Copyright (c) 2019, Mirco Schönfeld
+ *
+ *	 This source code is released for free distribution under the terms of the
+ *	 GNU General Public License version 2 or (at your option) any later version.
+ *
+ *	 This module contains functions for generating tags for TeX language files.
+ *
+ *	 Tex language reference:
+ *		 http://en.wikibooks.org/wiki/TeX#The_Structure_of_TeX
+ */
+
+/*
+ *	 INCLUDE FILES
+ */
+#include "general.h"	/* must always come first */
+#include <ctype.h>	/* to define isalpha () */
+#ifdef DEBUG
+#include <stdio.h>
+#endif
+#include <string.h>
+
+#include "debug.h"
+#include "entry.h"
+#include "keyword.h"
+#include "parse.h"
+#include "read.h"
+#include "routines.h"
+#include "vstring.h"
+
+/*
+ *	 MACROS
+ */
+#define isType(token,t)		(bool) ((token)->type == (t))
+#define isKeyword(token,k)	(bool) ((token)->keyword == (k))
+#define isIdentChar(c) (isalpha (c) || isdigit (c) || (c) == '_' || (c) == '-')
+
+/*
+ *	 DATA DECLARATIONS
+ */
+
+/*
+ * Used to specify type of keyword.
+ */
+enum eKeywordId {
+	KEYWORD_article,
+  KEYWORD_book,
+  KEYWORD_booklet,
+  KEYWORD_conference,
+  KEYWORD_inbook,
+  KEYWORD_incollection,
+  KEYWORD_inproceedings,
+  KEYWORD_manual,
+  KEYWORD_mastersthesis,
+  KEYWORD_misc,
+  KEYWORD_phdthesis,
+  KEYWORD_proceedings,
+  KEYWORD_techreport,
+  KEYWORD_unpublished
+};
+typedef int keywordId; /* to allow KEYWORD_NONE */
+
+enum eTokenType {
+	/* 0..255 are the byte's value.  Some are named for convenience */
+  TOKEN_COMMA = ',',
+	TOKEN_OPEN_CURLY = '{',
+	TOKEN_CLOSE_CURLY = '}',
+	/* above is special types */
+	TOKEN_UNDEFINED = 256,
+	TOKEN_KEYWORD,
+	TOKEN_IDENTIFIER,
+	TOKEN_STRING,
+};
+typedef int tokenType;
+
+typedef struct sTokenInfo {
+	tokenType		type;
+	keywordId		keyword;
+	vString *		string;
+	vString *		scope;
```

Comment:
It seems that scope field is not used.

Reply:
you're right. I removed the field in 952f651